docs/proposals/variable_changes.md

```diff
@@ -0,0 +1,308 @@
+r_openshift_hosted_registry_use_firewalld: "{{ os_firewall_use_firewalld | default(False) }}"
+
+openshift_hosted_router_wait: True
+openshift_hosted_registry_wait: True
+
+registry_volume_claim: 'registry-claim'
+
+openshift_hosted_router_edits:
+- key: spec.strategy.rollingParams.intervalSeconds
+  value: 1
+  action: put
+- key: spec.strategy.rollingParams.updatePeriodSeconds
+  value: 1
+  action: put
+- key: spec.strategy.activeDeadlineSeconds
+  value: 21600
+  action: put
+
+openshift_hosted_routers:
+- name: router
+  replicas: "{{ replicas | default(1) }}"
+  namespace: default
+  serviceaccount: router
+  selector: "{{ openshift_hosted_router_selector | default(None) }}"
+  images: "{{ openshift_hosted_router_image | default(None)  }}"
+  edits: "{{ openshift_hosted_router_edits }}"
+  stats_port: 1936
+  ports:
+  - 80:80
+  - 443:443
+  certificate: "{{ openshift_hosted_router_certificate | default({}) }}"
+
+openshift_hosted_router_certificate: {}
+openshift_hosted_registry_cert_expire_days: 730
+openshift_hosted_router_create_certificate: True
+
+r_openshift_hosted_router_os_firewall_deny: []
+r_openshift_hosted_router_os_firewall_allow: []
+
+r_openshift_hosted_registry_os_firewall_deny: []
+r_openshift_hosted_registry_os_firewall_allow:
+- service: Docker Registry Port
+  port: 5000/tcp
+cond: "{{ r_openshift_hosted_use_calico }}"
+```
+
+Except, it's not defined.  It's only referenced, and no default is provided.
+That means this role will fail without providing that value either through
+inventory, or as we're currently doing in openshift-ansible, via a play that
+provides the value explicitly.
+
+## Impact
+This means that this role is 100% dependent on a play passing in a value.
+It is currently not sufficient to provide a value in your inventory for this variable
+directly because as soon as you run one of the aforementioned plays, the value
+the user sets in their inventory is overridden.
+
+If you wanted this role to work both with and without our plays, you'd need to
+set the following in your inventory/group_vars:
+```yaml
+r_openshift_hosted_use_calico: True
+openshift_use_calico: True
+```
+The first line, because that's what the role's expecting, if you don't set it,
+the role will fail without our plays.
+The second line, because that's what openshift-ansible common expects and
+that is the value that is going to be provided to r_openshift_hosted_use_calico
+no matter what you set in the inventory for r_openshift_hosted_use_calico.
+
+Should this role fail without this specific variable?  In this case, no it shouldn't.
+Installing calico is a special case, not the default case.  The default value
+should definitely be false.
+
+Can this role exist in a vacuum?  Possibly, after we remove all the meta
+dependencies, this role might have meaningful use outside of openshift-ansible.
+I think it's desirable to strive for this goal.  But, in reality, this role
```

Comment:
+1 to this. I think our goal should be to create roles that _can_ be used independently, not just for external customers to consume them, but for our own consumption with our playbooks. We should want the roles to be just as easy for us to use as it is for customers.